pkg/river/parser/internal.go

```diff
@@ -638,6 +665,91 @@ var statementEnd = map[token.Token]struct{}{
 	token.COMMA:      {},
 }
 
+// parseInterpString parses an interpolated string sequence.
+func (p *parser) parseInterpString() ast.Expr {
```

Comment:
Were there any libraries that we considered using instead?

something like https://github.com/Knetic/govaluate? but there are probably better candidates out there too

Reply:
If there's a library out there that returns a parse tree for specifically the language we want to expose, and it's performant, then I wouldn't be against using a library.

Reply:
Well, obviously there wouldn't be one for River, but evaluating expressions inside strings, like `"1 + 2"` -> `3` could probably be found.

Reply:
Right, but the problem is the expression inside of string should be a River expression, including referencing component exports like `"${prometheus.exporter.unix.targets[0].__address__}" like in the example attached to the PR. 

That needs to be parsed and not directly evaluated so we can make sure dependencies are hooked up properly in the DAG.